packages/paywall/src/Paywall.ts

```diff
@@ -76,11 +76,20 @@ export class Paywall {
     this.networkConfigs = networkConfigs
     // Use provider in parameter, fall back to injected provider in window (if any)
     this.provider = provider || getProvider(window as Web3Window)
-    this.resetConfig(paywallConfig)
+    this.paywallConfig = injectProviderInfo(paywallConfig, this.provider)
     // Always do this last!
     this.loadCache()
   }
 
+  authenticate = () => {
+    if (this.iframe) {
+      this.showIframe()
+    } else {
+      this.shakeHands()
+    }
+    this.sendOrBuffer('authenticate', {})
```

Comment:
Does this not result in lag?

Someone starts with checkout iframe then show the oauth one. This.sendOrBuffer triggers the config change after showing the screen or is it not noticeable in practice?

Reply:
I have not seen any lag. I will do more testing.